### PR TITLE
add RepositoryModule.kt and RoomInventoryRepository.kt; Refactor Inve…

### PR DESCRIPTION
**Pull request by MaaxGr**
…ntoryViewModel.kt to Flow; Add AppResult.kt and collectFor() extension

### CODE REVIEW

File: app/src/main/java/de/madem/homium/utilities/extensions/AndroidCoreExtensions.kt [hunk 1]
```diff
@@ -266,4 +267,11 @@ fun Fragment?.whenSearchViewHandler(action: (SearchViewHandler) -> Unit){
     s.notNull {
         action.invoke(it)
     }
+}
+
+fun <T> Flow<T>.collectFor(viewLifecycleOwner: LifecycleOwner, onEmit: suspend (T) -> Unit) {
```
Comment (ericmuench) [inline app/src/main/java/de/madem/homium/utilities/extensions/AndroidCoreExtensions.kt]: Suggestion: Maybe we should call it `onCollect` instead. :D

File: app/src/main/java/de/madem/homium/databases/InventoryDao.kt [hunk 2]
```diff
@@ -4,13 +4,16 @@ import androidx.room.Dao
...
 import de.madem.homium.models.InventoryItem
-import java.util.*
+import kotlinx.coroutines.flow.Flow
 
 @Dao
 interface InventoryDao {
 
     @Query("SELECT * FROM inventoryItem")
-    fun fetchAllInventoryItems(): List<InventoryItem>
+    fun fetchAllInventoryItemsOld(): List<InventoryItem>
```
Comment (ericmuench) [inline app/src/main/java/de/madem/homium/databases/InventoryDao.kt:14]: Please mark this one with `@Deprecated` or a TODO to make sure that we don't forget the removal of this function

File: app/src/main/java/de/madem/homium/repositories/InventoryRepository.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import de.madem.homium.models.InventoryItem
+import kotlinx.coroutines.flow.Flow
+
+interface InventoryRepository {
+
+    fun getAllInventoryItems(): Flow<List<InventoryItem>>
+
+    //TODO: Add return type
+    fun deleteInventoryItemById(id: Int)
```
Comment (ericmuench) [inline app/src/main/java/de/madem/homium/repositories/InventoryRepository.kt]: Please make this Function a `suspend`-Function and call it in `viewModelScope.launch`

Comment (MaaxGr) [reply app/src/main/java/de/madem/homium/repositories/InventoryRepository.kt]: Resolved

File: app/src/main/java/de/madem/homium/databases/InventoryDao.kt (hunk 2 above)
Comment (MaaxGr) [reply app/src/main/java/de/madem/homium/databases/InventoryDao.kt:14]: Done

File: app/src/main/java/de/madem/homium/utilities/extensions/AndroidCoreExtensions.kt (hunk 1 above)
Comment (MaaxGr) [reply app/src/main/java/de/madem/homium/utilities/extensions/AndroidCoreExtensions.kt]: yeah probably a better name

**Pull request merged**